packages/compass/src/main/index.ts

```diff
@@ -108,7 +108,14 @@ async function main(): Promise<void> {
     });
   }
 
-  await CompassApplication.init(mode, globalPreferences);
+  try {
+    await CompassApplication.init(mode, globalPreferences);
+  } catch (e) {
+    // eslint-disable-next-line no-console
+    await handleUncaughtException(e as Error).then((e) => console.error(e));
+    // eslint-disable-next-line no-console
+    await CompassApplication.runExitHandlers().then((e) => console.error(e));
```

Comment:
The error will be written to the stdout by the `handleUncaughtException` method, and we probably want to exit after handlers are run, similar to how the `errorOutDueToAdditionalCommandLineFlags` handled a few lines above

```suggestion
    await handleUncaughtException(e as Error);
    await CompassApplication.runExitHandlers().finally(() => app.exit(1));
    return;
```